modules/component/src/core/utils/is-ivy.ts

```diff
@@ -0,0 +1,15 @@
+import { getGlobalThis } from './get-global-this';
+
+export function isIvy(): boolean {
+  const ng: any = getGlobalThis().ng;
+
+  // Is the global ng object is unavailable?
+  // ng === undefined in Ivy production mode
+  // View Engine has the ng object both in development mode and production mode.
+  return (
+    ng === undefined ||
+    // in case we are in dev mode in ivy
+    // `probe` property is available on ng object we use View Engine.
+    ng.probe === undefined
+  );
```

Comment:
these comments are hard to understand for me.
I think this would've helped me
```ts

// Table for ng global presence in ViewEngine and Ivy for prod/dev modes:
//
// | render     |  ViewEngine    |  ViewEngine    |      Ivy          |      Ivy          |     
// | mode       |     prod       |      dev       |      prod         |      dev          |
// | ng         |     present    |     present    |     undefined     |     present       |
// | ng.probe   |     present    |     present    |     undefined     |     undefined     |
//
// So for Ivy we need to make sure that ng is undefined or, 
// in case of dev environment, ng.probe is undefined

return ng?.probe === undefined 
```

Reply:
Added the comments.
https://github.com/ngrx/platform/pull/2046/commits/9181a10854ec7848bd4049996fbbfc7a47183b85
Thanks for the table!